### PR TITLE
[#1589] Fire PropertyChangeEvent only when value is changing

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@karakun.com>

### CODE REVIEW

**Comment by cdietrich** [comment]
please rebase

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java [hunk 1]
```diff
@@ -380,7 +380,7 @@ public void firePropertyChangeEvent(String name, Object oldValue,
 		// important: create intermediate array to protect against listeners
 		// being added/removed during the notification
 		final Object[] list = getListeners();
-		if (list.length == 0) {
+		if (list.length == 0 || Objects.equal(oldValue, newValue)) {
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java:383]: this differs from other pref stores impls. eg.
ScopedPreferenceStore where this seems to be c&ped from

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java:383]: Also my impression. If in doubt, better change the listener impl that does too many things when the event is indicating a no-op.

Please note that platform classes have logic like this:
```
if (changed > 0)
			masterPreferenceStore.firePropertyChangeEvent(getPreferencePrefix(), null, null);
```
This event would be swallowed with the proposed change.

**Review by szarnekow — CHANGES_REQUESTED** [review]
Please investigate further why you see an endless loop.

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java:383]: Thanks for the pointers. I'll have a second look.

**Comment by cdietrich** [comment]
@kthoms ok with move to 2.25?

**Comment by cdietrich** [comment]
move to 2.25

**Comment by cdietrich** [comment]
cleared milestone. feel free to readd if you are working on this again @kthoms

**Comment by kthoms** [comment]
Let's close this for now. I am not able to reproduce this ATM.

**Pull request closed without merging**